AccountingSystem/AccountingSystem/CSVHandler.h

```diff
@@ -0,0 +1,176 @@
+#pragma once
+#include <string>
+#include <functional>
+#include <vector>
+#include <cstdio>
+#include <experimental/filesystem>
+#include "IDataHandler.h"
+
+template <class T>
+class CSVHandler : public IDataHandler<T>
+{
+public:
+	 CSVHandler(std::string pathToFile, std::vector<std::string> columnsName, std::function<std::string(T)> convertToInputFormat,
+	            std::function<T(std::string)> convertToOutputFormat);
+	 int Add(const T data) override;
+	 int Delete(const int id) override;
+	 int Update(const int id, const T value) override;
+	 T Get(const int id) override;
+     T GetLastRecord() override;
+     int TableSize() override;
+private:
+	void CreateFile(const std::string pathToFile);
+    void CreateTmpFile(const std::string pathToFile, std::vector<std::string> rows);
+    std::string searchRow(int id);
+	std::function<std::string(T)> _toInputFormat;
```

Comment:
Optional but it's a little easier to read when you separate data and function into seprate blocks (with comments, empty line or another `private`)
```suggestion
    std::string searchRow(int id);
    
private:
    std::function<std::string(T)> _toInputFormat;
```